### PR TITLE
Remove protobuf dependency from Fetch method

### PR DESCRIPTION
**Pull request by willgreenberg**
Change the Fetch method signature to take strings for galaxy and key instead of protobuf messages, and update the HTTP implementation accordingly.

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -151,46 +149,34 @@ type httpFetcher struct {
...
 	defer res.Body.Close()
 	if res.StatusCode != http.StatusOK {
-		return fmt.Errorf("server returned: %v", res.Status)
+		return nil, fmt.Errorf("server returned: %v", res.Status)
 	}
-	b := bufferPool.Get().(*bytes.Buffer)
-	b.Reset()
-	defer bufferPool.Put(b)
-	_, err = io.Copy(b, res.Body)
+	data, err := ioutil.ReadAll(res.Body)
```
Comment (dfinkel) [inline http.go]: Now that we're reading the raw bytes of the value from the body of the HTTP response, we need to stop calling `proto.Marshal()` in the HTTP handler. on line 138: https://github.com/vimeo/groupcache/blob/86b85e8b6a8981735d6bc7f93f7f25f165bc0972/http.go#L138

We shouldn't have any `github.com/golang/protobuf/proto` or `github.com/vimeo/groupcache/groupcachepb` imports left in the main `galaxycache` package when you're done.

File: http.go [hunk 2]
```diff
@@ -151,46 +149,35 @@ type httpFetcher struct {
...
 	}
 	res, err := tr.RoundTrip(req)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer res.Body.Close()
 	if res.StatusCode != http.StatusOK {
-		return fmt.Errorf("server returned: %v", res.Status)
+		return nil, fmt.Errorf("server returned: %v", res.Status)
```
Comment (sachinagada) [inline http.go]: nit: can we specify in the error that it's an http response status code

Comment (sachinagada) [inline http.go]: nit: can we specify in the error that it's an http response status code

File: http.go [hunk 3]
```diff
@@ -137,60 +132,48 @@ func (h *HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Write the value to the response body as a proto message.
```
Comment (sachinagada) [inline http.go]: nit: outdated commment

Also on line 139, don't set header to protobuf since we no longer have the protobuf dependency

**Pull request merged**